test/e2e/framework/sources/sink.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright (c) 2020 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sources
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/framework/deployment"
+
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/ptr"
+)
+
+const (
+	eventDisplayName           = "event-display"
+	eventDisplayContainerImage = "gcr.io/knative-releases/knative.dev/eventing-contrib/cmd/event_display:latest"
```

Comment:
hopefully this image path with never break. I would be tempted to write a test to make sure the image is there.

Reply:
I believe we can rely on the side effect of the `WaitFor...` function to do that. If the image can't be pulled, the test will fail after a timeout, and the framework will surface the pod's failure reason in the logs.

We could also store the helper images our tests depend on in `gcr.io/triggermesh/e2e`, what do you think?

Reply:
if the error surfaced by `WaitFor` is meaningful than I am fine with that.